code/datums/position_point_vector.dm

```diff
@@ -0,0 +1,248 @@
+//Designed for things that need precision trajectories like projectiles.
+//Don't use this for anything that you don't absolutely have to use this with (like projectiles!) because it isn't worth using a datum unless you need accuracy down to decimal places in pixels.
+
+/proc/return_precise_position(atom/A)
+	return new /datum/position(A)
+
+/proc/return_precise_point(atom/A)
+	return new /datum/point(A)
+
+/proc/return_debug_point(atom/A)
+	var/datum/point/debug/debug = new(A)
+	debug.debug()
+	return debug
+
+/datum/position			//For positions with map x/y/z and pixel x/y so you don't have to return lists. Could use addition/subtraction in the future I guess.
+	var/x = 0
+	var/y = 0
+	var/z = 0
+	var/pixel_x = 0
+	var/pixel_y = 0
+
+/datum/position/proc/valid()
+	return x && y && z && !isnull(pixel_x) && !isnull(pixel_y)
+
+/datum/position/New(_x = 0, _y = 0, _z = 0, _pixel_x = 0, _pixel_y = 0)	//first argument can also be a /datum/point.

```

Comment:
you're passing an atom above, but have no case here to handle an atom.

Reply:
oof.

Reply:
this was resolved for anyone following along